Main.java

```diff
@@ -23,4 +23,21 @@ public static void main(String[] args){
 
     }
 
+    //Fills the FUV by reading the PUV and PUM
+    private void fillFUV(){
+        for(int i = 0; i < 15; i++){
+            if(!PUV[i]){ // If PUV[i] is false PUM[i] doesn't matter, FUV[i] is always true
+                FUV[i] = true;
+                continue;
+            } else { //otherwise makes sure everything in PUM[i] is true
```

Comment:
I think we need to set the default value to `true` otherwise we never do it when the whole row of the PUM is `true`.
```suggestion
            } else { //otherwise makes sure everything in PUM[i] is true
            FUV[i] = true;
```